src/mad_spark_alt/evolution/semantic_operators.py

```
@@ -11,15 +11,95 @@
 import random
 import time
 from datetime import datetime, timezone
-from typing import Any, Dict, List, Optional, Tuple
+from typing import Any, Dict, List, Optional, Tuple, Union
 
 from mad_spark_alt.core.interfaces import GeneratedIdea
 from mad_spark_alt.core.llm_provider import GoogleProvider, LLMRequest, LLMResponse
```

Ruff (0.12.2)

`mad_spark_alt.core.llm_provider.LLMResponse` imported but unused

Remove unused import: `mad_spark_alt.core.llm_provider.LLMResponse`

(F401)